app/controllers/sources_controller.rb

```diff
@@ -13,6 +14,8 @@ def index
 
   def show
     @source = Source.find(params[:id])
+    add_breadcrumb @source.source_set.name, source_set_path(@source.source_set)
+    add_breadcrumb @source.name, source_path(@source)
```

Comment:
Instead of the `@source.name` here, could we just have the text of the link be "Source"?  Some of the source names are several hundred characters long, and that's a lot to have in a breadcrumb.